clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTaskTest.java

```diff
@@ -2190,10 +2166,8 @@ public void testDataValidationCheckPointing(SortedInput sortedInput, AAConfig aa
       relevantPartitions.stream().forEach(partition -> {
         PubSubTopicPartition pubSubTopicPartition =
             new PubSubTopicPartitionImpl(pubSubTopicRepository.getTopic(topic), partition);
-        verify(mockLocalKafkaConsumer, timeout(LONG_TEST_TIMEOUT).atLeast(totalNumberOfConsumptionRestarts + 1))
+        verify(mockLocalKafkaConsumer, timeout(LONG_TEST_TIMEOUT).atLeast(1))
             .subscribe(eq(pubSubTopicPartition), anyLong());
-        verify(mockLocalKafkaConsumer, timeout(LONG_TEST_TIMEOUT).atLeast(totalNumberOfConsumptionRestarts))
-            .unSubscribe(eq(pubSubTopicPartition));
```

Comment:
The previous unsubscribe action is also done in the SIT thread, why that can complete but not with the "infiniteLockThread"?